### PR TITLE
Allow opt-in RBF using TransactionBuilder

### PR DESCRIPTION
**Pull request by lontivero**
This PR is for discussing a possible implementation to opt-in signaling `RBF` in transactions built using the `TransactionBuilder`. It adds a `TransactionBuilder.OptInRBF` property (default: `False`) that when set to true sets *all* the inputs' sequence to `0xfffffffd`.

### CODE REVIEW

File: NBitcoin/TransactionBuilder.cs [hunk 1]
```diff
@@ -1377,7 +1387,12 @@ bool IsEconomical(Coin c)
 				var input = ctx.Transaction.Inputs.FirstOrDefault(i => i.PrevOut == coin.Outpoint);
 				if(input == null)
 					input = ctx.Transaction.Inputs.Add(coin.Outpoint);
-				if(_LockTime != null && !ctx.NonFinalSequenceSet)
+
+				if(OptInRBF)
```
Comment (lontivero) [inline NBitcoin/TransactionBuilder.cs]: Here is where I am not 100% sure it is correct.

Comment (NicolasDorier) [reply NBitcoin/TransactionBuilder.cs]: It is not correct. Inverse the if and the if/else.
You are doing `if A => RBF elseif B => RBF+STUFF`

So if A is true, we are missing `STUFF`.
So you should do

`if B => RBF+STUFF elseif A =>RBF`

(Note that setting sequence to 0 effecitively put RBF on)

Comment (NicolasDorier) [reply NBitcoin/TransactionBuilder.cs]: Or remove the if else.

`if A => RBF; if B => RBF + STUFF`

File: NBitcoin/TransactionBuilder.cs [hunk 2]
```diff
@@ -1377,7 +1387,12 @@ bool IsEconomical(Coin c)
 				var input = ctx.Transaction.Inputs.FirstOrDefault(i => i.PrevOut == coin.Outpoint);
 				if(input == null)
 					input = ctx.Transaction.Inputs.Add(coin.Outpoint);
-				if(_LockTime != null && !ctx.NonFinalSequenceSet)
+
+				if(OptInRBF)
+				{
+					input.Sequence = Sequence.OptInRBF;
+				}
+				else if(_LockTime != null && !ctx.NonFinalSequenceSet)
```
Comment (NicolasDorier) [inline NBitcoin/TransactionBuilder.cs]: remove the else here, and I think it is good.

File: NBitcoin/TransactionBuilder.cs (hunk 1 above)
Comment (lontivero) [reply NBitcoin/TransactionBuilder.cs]: > (Note that setting sequence to 0 effectively activate RBF)
F*ck! You're right.

**Comment by lontivero** [comment]
Done. The failing test is `NBitcoin.Tests.ProtocolTests.CanGetChainsConcurrenty [FAIL]` on OSX.

**Pull request merged**